src/arch/mod.rs

```diff
@@ -29,11 +29,11 @@ pub trait Architecture {
 }
 
 pub trait ArchitectureMemory {
-    fn new<'alloc>(allocator: &mut mm::SimplePageAllocator<'alloc>) -> &'alloc mut Self;
+    fn new<'alloc>(allocator: &mut impl mm::PageAllocator) -> &'alloc mut Self;
```

Comment:
```suggestion
    fn new<'alloc>(allocator: &mut dyn mm::PageAllocator) -> &'alloc mut Self;
```
I think this is more idiomatic. References on trait objects should use `dyn`, while actual moved trait objects should use `impl`.

Reply:
cannot be done
```rust
error[E0038]: the trait `Architecture` cannot be made into an object
  --> src/mm/mod.rs:55:23
   |
55 |     pub fn new(arch: &dyn arch::Architecture) -> Self {
   |                       ^^^^^^^^^^^^^^^^^^^^^^ `Architecture` cannot be made into an object
   |
note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
```
And this is only the first 20% of the error message